woocommerce/woocommerce
# Remove leftovers from the reverted improved filtering mechanism for variations

### All Submissions:


### Changes proposed in this Pull Request:

Although the mechanism for improved filtering by attribute for variations [was reverted](https://github.com/woocommerce/woocommerce/pull/27625), there's some leftover code that still creates attribute-related term relationships for variations on save (and deletes them on product or variation deletion), also the database contains those no longer needed term relationships. This pull request:

1. Removes the code that creates/deletes product attribute attribute term relationships for variations.
2. Voids the original data migration that created these term relationships for already existing products.
3. Introduces a new data migration that deletes those no longer needed term relationships.

Closes #27651.

### How to test the changes in this Pull Request:

1. Without the fix, create a variable product with some variations, see that there are term relationships for the variations in the database (you can run the query used by the `wc_update_470_remove_term_relationships_for_variations` migration introduced in this pull request).
2. Change to this branch and run the migration, the same database query should now return no results.
3. Create again a variable product, the database query should still return no results.

### Other information:


### Changelog entry

> Dev - Remove leftover code and data from the reverted improvement for variations filtering by attribute.
